core-bundle/src/Resources/contao/modules/Module.php

```diff
@@ -492,6 +434,31 @@ protected function compileNavigationRow(PageModel $objPage, PageModel $objSubpag
 		return $row;
 	}
 
+	/**
+	 * Prepare menu item for usage in nav_* templates.
+	 *
+	 * @return array
+	 *
+	 * @internal use the parseTemplate hook if you want to alter the nav_* templates
+	 */
+	protected function compileMenuItem(ItemInterface $menuItem, string $subitems = ''): array
+	{
+		$item = $menuItem->getExtras();
+
+		$item['subitems'] = $subitems;
+		$item['isActive'] = $menuItem->isCurrent();
+		$item['link'] = $item['title'];
+		$item['title'] = StringUtil::specialchars($menuItem->getLinkAttribute('title'), true);
+		$item['pageTitle'] = StringUtil::specialchars($item['pageTitle'], true);
+		$item['description'] = str_replace(array("\n", "\r"), array(' ', ''), $item['description']);
+		$item['href'] = $menuItem->getUri();
+		$item['nofollow'] = (strncmp($menuItem->getLinkAttribute('rel', ''), 'noindex,nofollow', 16) === 0); // backwards compatibility
+		$item['target'] = ($v = $menuItem->getLinkAttribute('target')) ? " target=\"$v\"" : '';
+		$item['rel'] = ($v = $menuItem->getLinkAttribute('rel')) ? " rel=\"$v\"" : '';
```

Comment:
Btw. I think we can put even more things into `Module::compileMenuItem` that are mostly only related to the current navigation module of Contao. E.g. CSS class generation: you already provide the `pageModel` in the extras and thus you could also let `Module::compileMenuItem` set the appropriate CSS classes based on the `pageModel` instead of the `FrontendMenuBuilder`. Would make the `FrontendMenuBuilder` cleaner yet again :)